src/action/base/delete_user.rs

```diff
@@ -0,0 +1,152 @@
+use nix::unistd::User;
+use target_lexicon::OperatingSystem;
+use tokio::process::Command;
+use tracing::{span, Span};
+
+use crate::action::{ActionError, ActionErrorKind, ActionTag};
+use crate::execute_command;
+
+use crate::action::{Action, ActionDescription, StatefulAction};
+
+/**
+Delete an operating system level user
+*/
+#[derive(Debug, serde::Deserialize, serde::Serialize, Clone)]
+pub struct DeleteUser {
+    name: String,
+}
+
+impl DeleteUser {
+    #[tracing::instrument(level = "debug", skip_all)]
+    pub async fn plan(name: String) -> Result<StatefulAction<Self>, ActionError> {
+        let this = Self { name: name.clone() };
+
+        match OperatingSystem::host() {
+            OperatingSystem::MacOSX { .. } | OperatingSystem::Darwin => (),
+            _ => {
+                if !(which::which("useradd").is_ok() || which::which("adduser").is_ok()) {
```

Comment:
Why are we checking for add user commands in `DeleteUser`?

Reply:
Ahhhhh :astonished: Good catch.